pymtl3/dsl/test/ComponentLevel3_test.py

```diff
@@ -766,6 +775,108 @@ def construct( s ):
     return
   raise Exception("Should've thrown InvalidConnectionError.")
 
+def test_const_connect_struct_signal_to_struct():
+
+  @bitstruct
+  class SomeMsg1:
+    a: Bits8
+    b: Bits32
+
+  class Top( ComponentLevel3 ):
+    def construct( s ):
+      s.out = OutPort(SomeMsg1)
+      connect( s.out, SomeMsg1(1,3) )
+
+  x = Top()
+  x.elaborate()
+  simple_sim_pass(x)
+  x.tick()
+  assert x.out == SomeMsg1(1,3)
+
+def test_const_connect_nested_struct_signal_to_struct():
+
+  @bitstruct
+  class SomeMsg1:
+    a: Bits8
+    b: Bits32
+
+  @bitstruct
+  class SomeMsg2:
+    a: SomeMsg1
+    b: Bits32
+
+  class Top( ComponentLevel3 ):
+    def construct( s ):
+      s.out = OutPort(SomeMsg2)
+      connect( s.out, SomeMsg2(SomeMsg1(1,2),3) )
+
+  x = Top()
+  x.elaborate()
+  simple_sim_pass(x)
+  x.tick()
+  assert x.out == SomeMsg2(SomeMsg1(1,2),3)
+
+def test_const_connect_cannot_handle_same_name_nested_struct():
+
+  class A:
+    @bitstruct
+    class SomeMsg1:
+      a: Bits8
+      b: Bits32
+
+  class B:
+    @bitstruct
+    class SomeMsg1:
+      c: Bits8
+      d: Bits32
+
+  @bitstruct
+  class SomeMsg2:
+    a: A.SomeMsg1
+    b: B.SomeMsg1
+
+  class Top( ComponentLevel3 ):
+    def construct( s ):
+      s.out = OutPort(SomeMsg2)
+      connect( s.out, SomeMsg2(A.SomeMsg1(1,2),B.SomeMsg1(3,4)) )
```

Comment:
By the way, I'm thinking if we can organize most of our test cases by the test component used in it. To add translation test for the same feature I have to duplicate the test component again in another test file...

Reply:
I guess import a test component from DSL might not be a good idea, but reusing tests across backends should be good. This is because not all tests in DSL are Component test but ComponentLevel1,2,3 etc.

Reply:
Isn't a valid ComponentL1 test also a valid Component test?